doc/user/content/transform-data/optimization.md

```diff
@@ -13,52 +13,113 @@ aliases:
 * [JOIN](#join)
 * [DEFAULT](#default-index)
 
-`GROUP BY`, `ORDER BY` and `LIMIT` clauses currently do not benefit from an index.
+### `WHERE` point lookups
+
+Unlike some other databases, Materialize can use an index to serve query results
+even if the query does not specify a `WHERE` condition on the index key(s). For
+some queries, Materialize can perform [**point
+lookups**](/concepts/indexes/#point-lookups) on the index (as opposed to an
+index scan) if the query's `WHERE` clause:
 
+- Specifies equality (`=` or `IN`) condition on **all** the indexed fields. The
+  equality conditions must specify the **exact** index key expression (including
+  type).
 
-### `WHERE` point lookups
-Speed up a query involving a `WHERE` clause with equality comparisons to literals (e.g., `42`, or `'foo'`):
+- Only uses `AND` (conjunction) to combine conditions for **different** fields.
+
+Depending on your query pattern, you may want to build indexes to support point
+lookups.
+
+#### Create an index to support point lookups
+
+To [create an index](/sql/create-index/) to support [**point
+lookups**](/concepts/indexes/#point-lookups):
+
+```mzsql
+CREATE INDEX ON obj_name (<keys>);
+```
+
+- Specify **only** the keys that are constrained in the query's `WHERE` clause.
+  If your index contains keys not specified in the query's `WHERE` clause, then
+  Materialize performs a full index scan.
+
+- Specify all (or subset of) keys that are constrained in the query pattern's
+  `WHERE` clause. If the index specifies all the keys, Materialize performs a
+  point lookup only. If the index specifies a subset of keys, then Materialize
+  performs a point lookup on the index keys and then filters these results using
+  the conditions on the non-indexed fields.
+
+- Specify index keys that **exactly match** the column expressions in the
+  `WHERE` clause. For example, if the query specifies `WHERE quantity * price =
+  100`, the index key should be `quantity * price` and not `price * quantity`.
 
-| Clause                                            | Index                                    |
+- If the `WHERE` uses `OR` clauses and:
+
+  - The `OR` arguments constrain all the same fields (e.g., `WHERE (quantity = 5
+    AND price = 1.25) OR (quantity = 10 AND price = 1.25)`), create an index for
+    the constrained fields (e.g., `quantity` and `price`).
+
+  - The `OR` arguments constrain some of the same fields (e.g., `WHERE (quantity
+    = 5 AND price = 1.25) OR (quantity = 10 AND item = 'brownie)`), create an
+    index for the intersection of the constrained fields (e.g., `quantity`).
+    Materialize performs a point lookup on the indexed key and then filters the
+    results using the conditions on the non-indexed fields.
+
+  - The `OR` arguments constrain completely disjoint sets of fields (e.g.,
+    `WHERE quantity = 5 OR item = 'brownie'`), try to rewrite your query using a
+    `UNION` (or `UNION ALL`), where each argument of the `UNION` has one of the
```

Comment:
Maybe show this rewriting?

In this example, we might want point-lookup indexes on _both_ `quantity` and `item`, depending. It's a sensitive question, though---each index will hold a copy of the data in memory! The speed/memory trade-off should be discussed _somewhere_. Relatedly, if the memory impact of having both indexes is too high, you might want to take a more global look at _all_ of your queries to determine which index to build.

Reply:
Added in patch (stuck in the discussion here but also recognize that we'll need to discuss the tradeoff in multiple places in the doc)